flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/sink/DynamicTableSink.java

```diff
@@ -188,7 +188,11 @@ interface Context {
          * </ul>
          *
          * <p>Note: will always return empty for the delete statement because it has no column list.
+         *
+         * @deprecated use {@link
+         *     org.apache.flink.table.connector.sink.abilities.SupportsTargetColumnWriting} instead.
          */
+        @Deprecated(since = "2.1")
         Optional<int[][]> getTargetColumns();
```

Comment:
is there a case to want to see the target columns but not write to them? In which case we could keep this method?

Reply:
IMHO, connectors can customize this behavior by implementing the `SupportsTargetColumnWriting` interface defined in pr: https://github.com/apache/flink/pull/26361. Except connectors, I'm not aware of any other components need this information currently.